### ISSUE TITLE
Variable substitution in specific patterns

### ISSUE THREAD

**Issue by kevinvalk** [fluxcd/kustomize-controller#429]
I have an Nginx configuration that is loaded into a config map. I was using variable substitution functionality from Flux Kustomization and this caused the following error:
```console
2021-09-16T05:35:06.623Z error Kustomization/apps.flux-system - Reconciliation failed after 109.510145ms, next try in 1h0m0s var substitution failed for &#39;apps-configmap-ui-login-nginx-bfd45ggt26&#39;: YAMLToJSON: yaml: line 5: found unknown escape character
```

I could not find any `${...}` patterns in my nginx.conf file. However, I did identify the problematic line (see below). Upon removal of this nginx line, the variable substitution worked again.

```config
...
location ~* \.(?:ico|css|js|gif|jpe?g|png)
...
```

**Comment by JaneLiuL** [comment]
Yes. Kustomize-controller will replcae the variable begin with `$` or `${}`, if you don't want to be replace, you need to label wor annotatied with `kustomize.toolkit.fluxcd.io/substitute: disabled`  .

**Comment by kevinvalk** [comment]
According the documentation

> Note that if you want to avoid var substitutions in scripts embedded in ConfigMaps or container commands, you must use the format `$var` instead of `${var}`. All the undefined variables in the format `${var}` will be substituted with string empty, unless a default is provided e.g. `${var:=default}`.

Hence I would assume I can freely use `$xyz` in my yaml and configmap (from files) that should **not** be substituted. **And this seems to work fine**

---

I just updated the issue to better describe the actual problem.
